flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -300,8 +300,7 @@ protected void setBlocked(boolean isBlocked) {
 	private void notifyCreditAvailable() {
 		checkState(partitionRequestClient != null, "Tried to send task event to producer before requesting a queue.");
 
-		// We should skip the notification if this channel is already released.
-		if (!isReleased.get() && !isBlocked()) {
+		if (!isBlocked()) {
```

Comment:
What happens if `notifyCreditAvailable` is carried on after releasing? Add unit test for this?

Reply:
Because here, we could also release right after the check or any time before the handling of this call in the netty thread, the netty thread needs to handle this case anyway.

A new unit test for this specific scenario seems reasonable though.